### PR TITLE
feat: implement v2 create api subscriptions

### PR DESCRIPTION
**Pull request by jhaeyaert**
## Issue

https://gravitee.atlassian.net/browse/APIM-1933

## Description

This PR adds support for api subscription creation. It also makes a few minor renaming on the `Subscription` entity to align wording.

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-ccoerjamdf.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml [hunk 1]
```diff
@@ -814,6 +814,30 @@ paths:
...
                     $ref: "#/components/responses/Error"
+        post:
+            tags:
+                - API Subscriptions
+            summary: Subscribe to an API's plan
+            description: |-
+                Create a new subscription on a given API.
+                
+                User must have the API_SUBSCRIPTION[CREATE] permission.
+            operationId: createApiPlan
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/management-openapi-v2.yaml]:
```suggestion
            operationId: createApiSubscription
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiSubscriptionsResource_CreateTest.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+import io.gravitee.rest.api.model.parameters.Key;
+import io.gravitee.rest.api.model.parameters.ParameterReferenceType;
+import io.gravitee.rest.api.model.permissions.RolePermission;
+import io.gravitee.rest.api.model.permissions.RolePermissionAction;
+import io.gravitee.rest.api.service.common.GraviteeContext;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.core.Response;
+import org.junit.Test;
+
+public class ApiSubscriptionsResource_CreateTest extends ApiSubscriptionsResourceTest {
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiSubscriptionsResource_CreateTest.java:43]:
```suggestion
public class ApiSubscriptionsResource_CreateApiSubscriptionTest extends ApiSubscriptionsResourceTest {
```

**Review by phiz71 — COMMENTED** [review]
LGTM.
Could you change the operationId of the new resource and adapt the test class name accordingly?
Could you also declare the new tags ÀPI Subscriptions` in the tag list at the beginning of the openapi descriptor?

Thanks a lot 🙏🏻

File: gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiSubscriptionsResource_CreateTest.java (hunk 2 above)
Comment (jhaeyaert) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/test/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiSubscriptionsResource_CreateTest.java:43]: Let wait for a proper global decision regarding the naming. We will apply the rule once for all then 👍

**Review by phiz71 — APPROVED** [review]

**Pull request merged**